File: TrabajoI.C/Java/interfaz/ClaseC.java
package interfaz;

public class ClaseC implements Operable {
    String nombre;
    int edad;

    public ClaseC(String nombre) {   // Constructor 1
        this.nombre = nombre;
        System.out.println("Constructor sobrecargado 1 en C: " + nombre);
    }

    public ClaseC(String nombre, int edad) { // Constructor 2
        this.nombre = nombre;
        this.edad = edad;
        System.out.println("Constructor sobrecargado 2 en C: " + nombre + ", " + edad);
    }

    @Override
    public void operar() {
        System.out.println("Operando en C");
    }
}

File: TrabajoI.C/C#/interfaz/ClaseC.cs
public class ClaseC : IOperable
{
    private string nombre;
    private int edad;

    public C(string nombre)  // Versión 1
    {
        this.nombre = nombre;
        Console.WriteLine($"C sobrecargado (solo nombre): {nombre}");
    }

    public C(string nombre, int edad)  // Versión 2
    {
        this.nombre = nombre;
        this.edad = edad;
        Console.WriteLine($"C sobrecargado (nombre + edad): {nombre}, {edad}");
    }

    public void Operar()
    {
        Console.WriteLine("Operando en C");
    }
}
